### PR TITLE
Shipping Labels: validate shipping label address screen

### PR DESCRIPTION
**Pull request by pmusolino**
Part of #2973 

## Description
As part of Shipping Labels M2, we need to present the address validation form. In this PR, I implemented the screen that shows the basic fields and the confirmation button below. There is still no logic behind it and some UI customizations will be applied later. All the missing things will be handled in separate PRs. 

## Changes
- Now the field `address` in `ShippingLabelAddressVerification` model is declared as optional.
- Implemented the new `ShippingLabelAddressValidationViewController` which represents the validation screen for the address (origin and destination). It contains all the cell configuration for the address fields, the keyboard management, and the confirmation button. 
- Implemented the `ShippingLabelAddressValidationViewModel` which is useful for configuring `ShippingLabelAddressValidationViewController`. For the moment, it just returns the Sections without any logic.
- The `ShippingLabelFormViewController` is now able to present the `ShippingLabelAddressValidationViewController`.
- The models `ShippingLabelAddressVerification` and `ShippingLabelAddressValidationResponse` are now exposed throught Yosemite.
- Implemented a method in `ShippingLabelFormViewModel` that is useful for transforming an `Address` into a `ShippingLabelAddress`.

## Testing
1. Launch the app in debug mode (because of the feature flag).
2. Open an order detail.
3. Press the button "Create Shipping Label".
4. Press the button "Continue" in the Create Shipping Label form.
5. The new screen for validating an address should be presented.

## Screenshot

Update release notes:
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/Classes/ViewRelated/Orders/Order Details/Shipping Labels/Create Shipping Label Form/Shipping Address Validation/ShippingLabelAddressValidationViewController.swift
Comment (jaclync) [inline WooCommerce/Classes/ViewRelated/Orders/Order Details/Shipping Labels/Create Shipping Label Form/Shipping Address Validation/ShippingLabelAddressValidationViewController.swift]: nit: validation seems to be an implementation detail within the address form, how about `ShippingLabelAddressFormViewController`?

File: WooCommerce/Classes/ViewRelated/Orders/Order Details/Shipping Labels/Create Shipping Label Form/Shipping Address Validation/ShippingLabelAddressValidationViewController.swift
Comment (jaclync) [inline WooCommerce/Classes/ViewRelated/Orders/Order Details/Shipping Labels/Create Shipping Label Form/Shipping Address Validation/ShippingLabelAddressValidationViewController.swift]: nit: it felt a bit odd to me seeing `ShippingLabelAddressVerification` passed to the address form VC, since `ShippingLabelAddressVerification` is the type sent to the Yosemite/Networking layer for the API validation call. How about we just pass `type` and `address` here, and let the view model assemble to `ShippingLabelAddressVerification` when it's making the validation call?
```suggestion
    init(type: ShipType, address: ShippingLabelAddress?) {
```

File: WooCommerce/Classes/ViewRelated/Orders/Order Details/Shipping Labels/Create Shipping Label Form/Shipping Address Validation/ShippingLabelAddressValidationViewController.swift
Comment (jaclync) [inline WooCommerce/Classes/ViewRelated/Orders/Order Details/Shipping Labels/Create Shipping Label Form/Shipping Address Validation/ShippingLabelAddressValidationViewController.swift]: nit: we have a helper for this - `removeLastCellSeparator()` 😄

File: WooCommerce/Classes/ViewRelated/Orders/Order Details/Shipping Labels/Create Shipping Label Form/Shipping Address Validation/ShippingLabelAddressValidationViewController.swift
Comment (jaclync) [inline WooCommerce/Classes/ViewRelated/Orders/Order Details/Shipping Labels/Create Shipping Label Form/Shipping Address Validation/ShippingLabelAddressValidationViewController.swift]: nit: we probably don't need to call this since it's already called in `configureTableView`?

File: WooCommerce/Classes/ViewRelated/Orders/Order Details/Shipping Labels/Create Shipping Label Form/Shipping Address Validation/ShippingLabelAddressValidationViewController.swift
Comment (jaclync) [inline WooCommerce/Classes/ViewRelated/Orders/Order Details/Shipping Labels/Create Shipping Label Form/Shipping Address Validation/ShippingLabelAddressValidationViewController.swift]: nit: maybe `phone` to be consistent with `ShippingLabelAddress` and `Address`?

File: WooCommerce/Classes/ViewRelated/Orders/Order Details/Shipping Labels/Create Shipping Label Form/Shipping Address Validation/ShippingLabelAddressValidationViewController.swift
Comment (jaclync) [inline WooCommerce/Classes/ViewRelated/Orders/Order Details/Shipping Labels/Create Shipping Label Form/Shipping Address Validation/ShippingLabelAddressValidationViewController.swift]: nit: similar to above, maybe `phoneField*`?
```suggestion
        static let phoneField = NSLocalizedString("Phones", comment: "Text field phones in Shipping Label Address Validation")
        static let phoneFieldPlaceholder = NSLocalizedString("Optional", comment: "Text field placeholder in Shipping Label Address Validation")
```

File: WooCommerce/Classes/ViewRelated/Orders/Order Details/Shipping Labels/Create Shipping Label Form/ShippingLabelFormViewModel.swift
Comment (jaclync) [inline WooCommerce/Classes/ViewRelated/Orders/Order Details/Shipping Labels/Create Shipping Label Form/ShippingLabelFormViewModel.swift]: nit: instead of exposing `Order` and this function, could the view model provide the origin and destination address of type `ShippingLabelAddress` directly?

in either case, the new API can be unit tested? 🙂

File: WooCommerce/Classes/ViewRelated/Orders/Order Details/Shipping Labels/Create Shipping Label Form/ShippingLabelFormViewModel.swift
Comment (jaclync) [inline WooCommerce/Classes/ViewRelated/Orders/Order Details/Shipping Labels/Create Shipping Label Form/ShippingLabelFormViewModel.swift]: this naming rule wouldn't work for certain languages, as the order is often reversed in a few Asian languages. if we want to support localized name correctly, we probably want to follow:

https://github.com/woocommerce/woocommerce-ios/blob/6fef0340b60f8819cb86a7663afd5eee0997d7ba/WooCommerce/Classes/Model/Address+Woo.swift#L13-L17

File: WooCommerce/Classes/ViewRelated/Orders/Order Details/Shipping Labels/Create Shipping Label Form/Shipping Address Validation/ShippingLabelAddressValidationViewModel.swift
Comment (jaclync) [inline WooCommerce/Classes/ViewRelated/Orders/Order Details/Shipping Labels/Create Shipping Label Form/Shipping Address Validation/ShippingLabelAddressValidationViewModel.swift]: nit: similar to the previous comment, maybe the view model can just provide `type` and `address` separately, since `ShippingLabelAddressVerification` is an implementation detail for the API validation call?

**Review by jaclync — APPROVED** [review]
Nice work on the address form! Just some suggestions on the code, testing is 👍

File: WooCommerce/Classes/ViewRelated/Orders/Order Details/Shipping Labels/Create Shipping Label Form/Shipping Address Validation/ShippingLabelAddressValidationViewController.swift
Comment (pmusolino) [reply WooCommerce/Classes/ViewRelated/Orders/Order Details/Shipping Labels/Create Shipping Label Form/Shipping Address Validation/ShippingLabelAddressValidationViewController.swift]: It makes totally sense 👍

File: WooCommerce/Classes/ViewRelated/Orders/Order Details/Shipping Labels/Create Shipping Label Form/ShippingLabelFormViewModel.swift
Comment (pmusolino) [reply WooCommerce/Classes/ViewRelated/Orders/Order Details/Shipping Labels/Create Shipping Label Form/ShippingLabelFormViewModel.swift]: Wow, definitely TIL something new. 💯 
I made all the changes related to this suggestion here 51a3d8630f47b9f8d5978eaa1a609047b786d189

**Comment by pmusolino** [comment]
Thank you for your review @jaclync 💯 I made all the changes and I'm waiting for the CI before merging it.

**Pull request merged**
